pkg/apis/awsprovider/v1alpha1/awsprovider_validation.go

```diff
@@ -0,0 +1,32 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+
+	"knative.dev/pkg/apis"
+)
+
+func (a *AWSProvider) Validate(ctx context.Context) (errs *apis.FieldError) {
+	return errs.Also(
+		apis.ValidateObjectMetadata(a).ViaField("metadata"),
+		a.Spec.validate(ctx).ViaField("spec"),
+	)
+}
+
+func (a *AWSProviderSpec) validate(ctx context.Context) (errs *apis.FieldError) {
```

Comment:
There's not a lot useful you can do here since the rest of the AWS Provider spec is in a different CRD. (I can't validate my UserData based off AMIFamily because the latter lives elsewhere).